### PR TITLE
MTDSA-17107 API can now pass correct 'intent' header to stub

### PR DESCRIPTION
**Pull request by sean-flynn-HMRC**

### CODE REVIEW

File: test/v1/connectors/DeleteCharitableGivingTaxReliefConnectorSpec.scala [hunk 1]
```diff
@@ -44,22 +45,42 @@ class DeleteCharitableGivingTaxReliefConnectorSpec extends ConnectorSpec {
         result shouldBe expectedOutcome
       }
 
-      "given a TYS request" in new TysIfsTest with Test {
+      "given a TYS request" when {
```
Comment (jeremystone) [inline test/v1/connectors/DeleteCharitableGivingTaxReliefConnectorSpec.scala:48]: Unfortunately the tests pass whether or not the 'intent' header is sent.

Comment (jeremystone) [reply test/v1/connectors/DeleteCharitableGivingTaxReliefConnectorSpec.scala:69]: This should be addressed by latest commit.

Comment (harry-ritchie) [reply test/v1/connectors/DeleteCharitableGivingTaxReliefConnectorSpec.scala:69]: Would it add value to the tests if we test the failed case when the feature is on but the header is excluded (+ vice versa)?

File: app/api/connectors/BaseDownstreamConnector.scala [hunk 2]
```diff
@@ -28,12 +28,11 @@ import scala.concurrent.{ExecutionContext, Future}
 trait BaseDownstreamConnector extends Logging {
   val http: HttpClient
   val appConfig: AppConfig
+  val featureSwitches: FeatureSwitches
```
Comment (harry-ritchie) [inline app/api/connectors/BaseDownstreamConnector.scala:31]: What's the value of adding the feature switch declaration here? I can't see where it gets used, only in each connector module as an implicit. Does it need to be in the interface?

File: app/v1/connectors/DeleteCharitableGivingTaxReliefConnector.scala [hunk 3]
```diff
@@ -37,14 +39,16 @@ class DeleteCharitableGivingTaxReliefConnector @Inject() (val http: HttpClient,
 
     import request._
 
+    val intent = if (featureSwitches.isPassDeleteIntentEnabled) Some("DELETE") else None
+
     if (taxYear.useTaxYearSpecificApi) {
       val downstreamUri = TaxYearSpecificIfsUri[Unit](s"income-tax/${taxYear.asTysDownstream}/$nino/income-source/charity/annual")
-      delete(downstreamUri)
-
+      delete(downstreamUri, intent = intent)
```
Comment (harry-ritchie) [inline app/v1/connectors/DeleteCharitableGivingTaxReliefConnector.scala:46]: Do we need to explicitly say `intent = intent`? Could we just put `intent`?

File: app/api/connectors/BaseDownstreamConnector.scala (hunk 2 above)
Comment (jeremystone) [reply app/api/connectors/BaseDownstreamConnector.scala:31]: Good spot! I will remove (and remove the unnecessary `val`s from each of the connectors).

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**